public/js/components/Scopes.js

```diff
@@ -71,15 +73,99 @@ function getScopes(pauseInfo) {
       scopes.push([
         scope.getIn(["object", "class"]),
         fromJS({
+          isScope: true,
           value: scope.get("object")
         })
       ]);
     }
+    index++;
   } while (scope = scope.get("parent")); // eslint-disable-line no-cond-assign
 
   return scopes;
 }
 
+function isObject(obj) {
+  return obj.getIn(["value", "type"]) === "object";
+}
+
+function loadedProperties(obj, loadedObjects) {
+  return loadedObjects.get(obj.getIn(["value", "actor"])) ||
+         obj.get("prefetchedProperties");
+}
+
+function sortProperties(a, b) {
+  const obj = isObject(a[1]) && !isObject(b[1]) ? -10 : 10;
+  const alpha = a[0] < b[0] ? -1 : 1
+
+  return obj + alpha > 0 ? -1 : 1;
```

Comment:
I think it look better with the way that we indent the tree, but I would prefer sorting them alphabetically. As a user, it's confusing if I'm looking for a property and have to remember if it's an object or not.

What we should fix is how we display the tree so that they don't look so out of place. It's problematic anyway because it's unclear which items are children; for example, in the above screenshot, is `a: 2` a top-level item or a child of `pythagorean`? It lines up perfectly with the other top-level items, so it looks like it's top-level. But it's actually a child of `pythagorean`. The way to fix this is to indent all the children more and make them all line up (whether of not they have arrows). If you look at Firefox/Chrome, that's exactly what they do.

So I'd prefer to revert this change and I can work on styling them. If you're not working on this today/tomorrow (because of conference) I'll merge this in and do a follow-up. I'll work on scopes later this week.